### PR TITLE
Refine and clean up runroot

### PR DESCRIPTION
**Pull request by chitianhao**
1. Some code cleaning up of runroot engine and test.
2. Improved functionality of runroot_handler with executables using runroot directly by finding the current working directory or executable path.

@dragon512

### CODE REVIEW

**Comment by zwoop** [comment]
[approve ci].

File: cmd/traffic_layout/engine.cc [hunk 1]
```diff
@@ -164,7 +163,7 @@ bool
 RunrootEngine::runroot_parse()
 {
   int i = 1;
```
Comment (dragon512) [inline cmd/traffic_layout/engine.cc]: instead of casting... just change the type to unsigned

**Comment by zwoop** [comment]
[approve ci debian]

**Comment by zwoop** [comment]
[approve ci centos]

**Comment by bryancall** [comment]
[approve ci debian]

File: cmd/traffic_layout/engine.cc [hunk 2]
```diff
@@ -164,7 +163,7 @@ bool
...
-  while (i < _argc) {
+  while (i < static_cast<int>(_argv.size())) {
```
Comment (zwoop) [inline cmd/traffic_layout/engine.cc]: Also, why not

```
while (unsigned i < ....
```

File: cmd/traffic_layout/engine.cc (hunk 1 above)
Comment (zwoop) [reply cmd/traffic_layout/engine.cc]: I meant, why not for (unsigned i=1; ...

**Comment by dragon512** [comment]
[approve ci]

**Comment by dragon512** [comment]
Known issues in testing this is a permission issue depending on location of original install.. to be fixed in a different PR that should add some verify and fix options

**Comment by dragon512** [comment]
[approve ci]

**Pull request merged**